### ISSUE TITLE
Support dynamic router base path

### ISSUE THREAD

**Issue by michael42** [nuxt/nuxt.js#7287]
### What problem does this feature solve?

We're currently building a universal Nuxt-based web app that will be available via multiple reverse proxies with distinct URLs, for example https://project1.com/abc/def and https://project2.com/def/qwerty. That doesn't work out of the box, because Nuxt configures the vue-router with a static base path, so all the reverse proxies would need to pass the user-visible URL as-is.

As a workaround, we're currently configuring Nuxt with `router: { base: '/' }`, let the reverse proxy rewrite the `<base href>` tag and use a patch to let the router actually use rewritten URL (not that this doesn't support IE11 and is probably not the best way to do it anyway).

```
diff --git a/node_modules/@nuxt/vue-app/template/router.js b/node_modules/@nuxt/vue-app/template/router.js
index bcbc109..24627cb 100644
--- a/node_modules/@nuxt/vue-app/template/router.js
+++ b/node_modules/@nuxt/vue-app/template/router.js
@@ -93,7 +93,9 @@ Vue.use(Router)
 
 export const routerOptions = {
   mode: '<%= router.mode %>',
-  base: decodeURI('<%= router.base %>'),
+  base: process.client ? new URL(document.baseURI).pathname : decodeURI('<%= router.base %>'),
   linkActiveClass: '<%= router.linkActiveClass %>',
   linkExactActiveClass: '<%= router.linkExactActiveClass %>',
   scrollBehavior,
```

### What does the proposed changes look like?

I think Nuxt should automatically use the configured `<base href="...">` to initialize the vue-router instead of the build-time constant from the `nuxt.config.js`.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10587">#c10587</a>)</em></sub></div>

**Comment by pimlie** [comment]
I think work on this is already in progress here: https://github.com/nuxt/nuxt.js/pull/7312 (/cc @pi0)

**Comment by pi0** [comment]
@pimlie Indeed. With runtimeConfig we can support both dynamic base for webpack and router.

**Comment by benjiwolff** [comment]
Hi, from my attempts it doesn't seem like runtimeConfig provides a dynamic router base path for statically generated apps. The issue seems to be that `app.baseURL` is evaluated at build time and so while you can fix API-calls with an environment variable/runtime config, nuxt still tries to fetch the bundled app from paths calculated at build time (e.g. `/_nuxt/Dk6n2bF3.js` etc.).

**Comment by danielroe** [comment]
this isn't the case. we have tests that verify the behaviour.

you set it at runtime with NUXT_APP_BASE_URL.

feel free to open a new issue with a reproduction 🙏

**Comment by benjiwolff** [comment]
I am talking about [statically hosted apps](https://nuxt.com/docs/4.x/getting-started/deployment#static-hosting) here. I don't see, how an environment variable in the server would affect the statically served html/javascript/css. There would need to be an explicit mechanism that informs the static app about where it is hosted.

**Comment by danielroe** [comment]
you are right that dynamic router base doesn't support prerendered apps. you would have to update the html to reference the correct paths.
